core/src/main/java/org/apache/calcite/rel/metadata/ReflectiveRelMetadataProvider.java

```diff
@@ -78,7 +78,9 @@ protected ReflectiveRelMetadataProvider(
       ConcurrentMap<Class<RelNode>, UnboundMetadata> map,
       Class<? extends Metadata> metadataClass0,
       Multimap<Method, MetadataHandler> handlerMap) {
-    assert !map.isEmpty() : "are your methods named wrong?";
+    if (map.isEmpty()) {
+      throw new IllegalArgumentException("ReflectiveRelMetadataProvider methods map is empty");
```

Comment:
Sorry I missed your PR. 

Why not use `CheckArgument` in Guava?

Reply:
Thanks for the feedback @amaliujia , you are right, I am considering open a follow-up PR to change it into `Preconditions.checkArgument`

Reply:
@amaliujia please check https://github.com/apache/calcite/pull/2100